substrate/client/rpc-spec-v2/src/chain_head/chain_head_follow.rs

```diff
@@ -705,71 +709,82 @@ where
 	async fn submit_events<EventStream>(
 		&mut self,
 		startup_point: &StartupPoint<Block>,
-		mut stream: EventStream,
+		stream: EventStream,
 		sink: Subscription,
 		rx_stop: oneshot::Receiver<()>,
 	) -> Result<(), SubscriptionManagementError>
 	where
-		EventStream: Stream<Item = NotificationType<Block>> + Unpin,
+		EventStream: Stream<Item = NotificationType<Block>> + Unpin + Send,
 	{
-		let mut stream_item = stream.next();
-
-		// The stop event can be triggered by the chainHead logic when the pinned
-		// block guarantee cannot be hold. Or when the client is disconnected.
-		let connection_closed = sink.closed();
-		tokio::pin!(connection_closed);
-		let mut stop_event = futures_util::future::select(rx_stop, connection_closed);
-
-		while let Either::Left((Some(event), next_stop_event)) =
-			futures_util::future::select(stream_item, stop_event).await
-		{
-			let events = match event {
+		// make the stream abortable
+		let (stream, cancel_handle) = futures::stream::abortable(stream);
```

Comment:
I don't really understand why having this abortable(stream) is needed because we would just poll it until either the stream returns an error or the subscription closed but doesn't hurt I guess.... or am I missing something?

Reply:
We need to stop the stream when `rx_stop` resolves

Reply:
hmm I don't follow after `rx_stop` resolves we will just drop stream again and never poll it again?

AFAIU, the abortable(stream) is just wrapper around the stream and simply returns None when it's polled after `stream.abort() is called` and doesn't poll the underlying stream again after that. What am I missing? :)

Reply:
Yup, we would never poll it, or I am missing something?

Reply:
so what's the point of having the abortable stream then? :thinking:

Reply:
Ahh, got it

Reply:
removed abortable